open_prices/common/authentication.py

```diff
@@ -13,22 +13,32 @@ def create_token(user_id: str) -> str:
     return f"{user_id}__U{str(uuid.uuid4())}"
 
 
-def get_authorization_token(authorization: str) -> str:
-    return authorization.split(" ")[1]
+def get_token_from_cookie(request: Request) -> str | None:
+    return request.COOKIES.get(settings.SESSION_COOKIE_NAME)
 
 
-def get_request_session(request: Request):
+def get_token_from_header(request: Request) -> str | None:
     authorization = request.META.get("HTTP_AUTHORIZATION")  # "Bearer <token>"
-    session_cookie = request.COOKIES.get(settings.SESSION_COOKIE_NAME)
+    if authorization and "__U" in authorization:
+        return authorization.split(" ")[1]
+    return None
+
+
+def has_token_from_cookie_or_header(request: Request) -> bool:
```

Comment:
This function returns a string or None, not a bool.
It still works, but for type consistency you should enclose everything under a `bool(...)`

```
>>> ("" or "test") == "test"
True
```

Reply:
I fixed it and added a bunch of tests in https://github.com/openfoodfacts/open-prices/pull/969/commits/ebfa1f35fea714f4b114cf128bd760e95a8a941d